pkg/git/gogit_test.go

```diff
@@ -84,6 +89,31 @@ var _ = Describe("Clone", func() {
 	})
 })
 
+var _ = Describe("ValidateAccess", func() {
+	It("validate access to a given repository successfully", func() {
+		err := gitClient.ValidateAccess(context.Background(), "https://github.com/githubtraining/hellogitworld", "master")
+		Expect(err).ShouldNot(HaveOccurred())
+	})
+
+	It("should not fail on an empty repo", func() {
+
+		fakeGoGit := &gitfakes.FakeGoGitI{}
+
+		fakeGoGit.PlainCloneContextReturns(nil, transport.ErrEmptyRemoteRepository)
+
+		fakeGitClient := git.New(nil, fakeGoGit)
+
+		err := fakeGitClient.ValidateAccess(context.Background(), "https://github.com/githubtraining/hellogitworld", "master")
+
+		Expect(err).ShouldNot(HaveOccurred())
+	})
+
+	It("fails to validate access to a possible private repository", func() {
+		err := gitClient.ValidateAccess(context.Background(), "https://github.com/notexisted/repo", "master")
```

Comment:
I get this test, but you are testing the internal of `PlainCloneContextReturns`.  At this point you could use the fake and return any error that isn't `transport.ErrEmptyRemoteRespository` and return

`fakeGoGit.PlainCloneContextReturns(nil, errors.New("ka-boom")`

That way if gogit changes internally the test is still valid.  Of course, if you want to return an error specifically about auth then that is a different case on my point above is moot.

Reply:
yes, I'm trying to test an auth error.